doc/sphinx/README.rst

```diff
@@ -0,0 +1,298 @@
+      matching :n:`@pattern` in the current goal.
+
+      .. exn:: Too few occurrences
+
+Objects are automatically collected into indices, and can be linked to using the role version of the object's directive. For example, you could link to the tactic variant above using ``:tacv:`simpl_at```, and to its exception using ``:exn:`Too few occurrences```.
+
+Names (link targets) are auto-generated for most simple objects, though they can always be overwritten using a `:name:` option, as shown above.
+
+- Options, errors, warnings have their name set to their signature, with ``...`` replacing all notation bits.  For example, the auto-generated name of ``.. exn:: @qualid is not a module`` is ``... is not a module``, and a link to it would take the form ``:exn:`... is not a module```.
+- Vernacs (commands) have their name set to the first word of their signature.  For example, the auto-generated name of ``Axiom @ident : @term`` is ``Axiom``, and a link to it would take the form ``:cmd:`Axiom```.
+- Vernac variants, tactic notations, and tactic variants do not have a default name.
+
+Notations
+---------
+
+The signatures of most objects can be written using a succinct DSL for Coq notations (think regular expressions written with a Lispy syntax).  A typical signature might look like ``Hint Extern @num {? @pattern} => @tactic``, which means that the ``Hint Extern`` command takes a number (``num``), followed by an optional pattern, and a mandatory tactic.  The language has the following constructs (the full grammar is in `TacticNotations.g <https://github.com/coq/coq/blob/master/doc/tools/coqrst/notations/TacticNotations.g>`):
+
+``@…``
+  A placeholder (``@id``, ``@num``, ``@tactic``)
+
+``{? …}``
+  an optional block
+
+``{* …}``, ``{+ …}``
+  an optional (``*``) or mandatory (``+``) block that can be repeated, with repetitions separated by spaces
+
+``{*, …}``, ``{+, …}``
+  an optional or mandatory repeatable block, with repetitions separated by spaces
+
+``%|``, ``%{``, …
+  an escaped character (rendered without the leading ``%``)
+
+..
+   FIXME document the new subscript support
+
+As an exercise, what do the following patterns mean?
+
+.. code::
+
+   pattern {+, @term {? at {+ @num}}}
+   generalize {+, @term at {+ @num} as @ident}
+   fix @ident @num with {+ (@ident {+ @binder} {? {struct @ident'}} : @type)}
+
+Objects
+-------
+
+Here is the list of all objects of the Coq domain (The symbol :black_nib: indicates an object whose signature can be written using the notations DSL):
+
+``.. cmd::`` :black_nib: A Coq command.
+    Example::
+
+       .. cmd:: Infix "@symbol" := @term ({+, @modifier}).
+
+          This command is equivalent to :n:`…`.
+
+``.. cmdv::`` :black_nib: A variant of a Coq command.
+    Example::
+
+       .. cmd:: Axiom @ident : @term.
+
+          This command links *term* to the name *ident* as its specification in
+          the global context. The fact asserted by *term* is thus assumed as a
```

Comment:
Are you sure this shouldn't be:
```
:token:`term`
```
and
```
:token`ident`
```
instead?

Reply:
Yup, of course. I copied the examples directly from the manual, and I didn't pay enough attention to the actual contents :/ Fixed here and there.